dotnet/src/Experimental/Process.Abstractions/KernelProcessStepState.cs

```diff
@@ -1,22 +1,41 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
 namespace Microsoft.SemanticKernel;
 
 /// <summary>
 /// Represents the state of an individual step in a process.
 /// </summary>
+[DataContract]
+[KnownType(nameof(GetKnownTypes))]
 public record KernelProcessStepState
 {
+    /// <summary>
+    /// A set of known types that may be used in serialization.
+    /// </summary>
+    private readonly static HashSet<Type> s_knownTypes = [];
+
+    /// <summary>
+    /// Used to dynamically provide the set of known types for serialization.
+    /// </summary>
+    /// <returns></returns>
+    private static HashSet<Type> GetKnownTypes() => s_knownTypes;
```

Comment:
Managing this as a dynamic set, I wonder about the the long-term consistency contract.  I suppose as long as every builder faithfully registeres types...still...static state such as this can often become problematic.

Reply:
I agree, it's unfortunate that it needs to work this way. For now we are locked in to this because Dapr actors still use DataContract serialization. They are actively working on supporting System.Text.Json and at that point we can make this better.